LDAP/ldap2s3/ldap2s3.py

```diff
@@ -179,52 +261,39 @@ def group(self, entry):
     users = {}
     sgen = mozldap.conn.extend.standard.paged_search(search_base=config.ldap.search_base.users,
                                                      search_filter=config.ldap.filter.users,
-                                                     attributes = ['mail', 'sshPublicKey', 'pgpFingerprint'],
+                                                     attributes = ['mail', 'sshPublicKey', 'pgpFingerprint', 'sn',
+                                                                   'givenName', 'workdayTimezone', 'telephoneNumber',
+                                                                   'createTimestamp', 'modifyTimestamp'],
                                                      search_scope=SUBTREE, paged_size=10, generator=True)
+    # Create the list of all users
     for entry in sgen:
-        u = mozldap.user(entry)
-        users[u.dn] = {'mail': u.mail, 'SSHFingerprints': u.sshPublicKey, 'PGPFingerprints': u.pgpFingerprint}
-
-    # Intersect all this to find which users belong to which group
-    # Users = {'dn here': 'mail value here', ...}
-    # Groups = {'group name here': ['dn here', ...], ...}
-
-    # Create the list of users per groups. We don't actually currently use this.
-#    grouplist = {}
-#    set_userskey = set(users.keys())
-#    for group in groups:
-#        uing = set(groups[group]) & set_userskey
-#        for u in uing:
-#            if not grouplist.get(group):
-#                grouplist[group] = []
-#            grouplist[group].append(users[u])
-
-    # Same but reverse (find which group belongs to which users). We actually use this ;-)
-    userlist = {}
-    # Prefill so that we include users with no group data and default other attributes
-    for u in users:
-        user = users[u]
-        useremail = user.get('mail')
-        userlist[useremail] = {'groups': [], 'SSHFingerprints': user.get('SSHFingerprints'), 'PGPFingerprints': user.get('PGPFingerprints')}
+        dn, u = mozldap.user(entry)
+        users[dn] = u
+
+    # Find which group belongs to which users and add them
     set_userskey = set(users.keys())
     for group in groups:
         uing = set(groups[group]) & set_userskey
         for u in uing:
-            useremail = users[u]['mail']
-            userlist[useremail]['groups'].append(group)
-
-    # Add user attributes
+            users[u]['groups'].append(group)
 
     logger.debug('Resulting group list:')
-    logger.debug(json.dumps(userlist))
-    userlist_json_str = json.dumps(userlist, ensure_ascii=False).encode('utf-8')
+    logger.debug(json.dumps(users))
+
+    # Validate all user profiles, warn strongly on failure
+    for u in users:
+        if not mozldap.validate_user(users[u]):
+            logger.critical("Profile schema validation failed for user {} - skipped".format(u))
```

Comment:
@jdow this is the actually critical validation step, because if this does not cause actual alerts, profiles don't get properly updated

Reply:
Agreed, I'm not sure the best way to pay attention to this. Cronmail is fine for stuff that can be dealt with within a week or two, but I think for something like this, we need a better alerting infrastructure, perhaps nagios or mozdef or something?

Reply:
cloudwatch event to slack?